tests/test_compiler.py

```diff
@@ -246,7 +246,7 @@ def circuit(x: float):
             return qml.expval(qml.PauliZ(0))
 
         mlir_str = str(circuit.mlir)
-        result_header = "func.func private @circuit(%arg0: tensor<f64>) -> tensor<f64>"
+        result_header = "func.func public @circuit(%arg0: tensor<f64>) -> tensor<f64>"
```

Comment:
Isn't it `func.func public @jit_circuit(%arg0: tensor<f64>) -> tensor<f64>`? 
https://github.com/PennyLaneAI/pennylane/actions/runs/11236559516/job/31237598930?pr=6282

I know the nested modules are then inlined, but the test doesn't seem to reach that stage? (i.e. in the output the module is still there)

Reply:
`jit_circuit` is the entry point to the host program, and `circuit` is the entry point to the qnode. I think this test was testing the entry to the qnode. However, the error message shows `jit_circuit` because it is the first function.